internal/nodes/avi_model_advl4_translator.go

```diff
@@ -147,11 +147,37 @@ func (o *AviObjectGraph) ConstructSvcApiL4VsNode(gatewayName, namespace, key str
 		gw, _ := lib.GetSvcAPIInformers().GatewayInformer.Lister().Gateways(namespace).Get(gatewayName)
 
 		var serviceNSNames []string
+		listenerSvcMapping := make(map[string][]string)
 		if found, services := objects.ServiceGWLister().GetGwToSvcs(namespace + "/" + gatewayName); found {
 			for svcListener, service := range services {
 				// assume it to have only a single backend service, the check is in isGatewayDelete
 				if utils.HasElem(listeners, svcListener) && len(service) == 1 && !utils.HasElem(serviceNSNames, service[0]) {
 					serviceNSNames = append(serviceNSNames, service[0])
+					if val, ok := listenerSvcMapping[svcListener]; ok {
+						listenerSvcMapping[svcListener] = append(val, service[0])
+					} else {
+						listenerSvcMapping[svcListener] = []string{service[0]}
+					}
+				}
+			}
+		}
+
+		var fqdns []string
+		for _, listener := range gw.Spec.Listeners {
+			autoFQDN := false
+			// Honour the hostname if specified corresponding to the listener.
+			if listener.Hostname != nil && string(*listener.Hostname) != "" {
+				fqdns = append(fqdns, string(*listener.Hostname))
+				autoFQDN = false
```

Comment:
is there a point of setting this variable to false again? It is already false at Line 167 right?

Reply:
thanks. Changed the initialization to `true`.